clients/spark/core/src/test/scala/io/treeverse/jpebble/ByteBufferIndexedBytesSpec.scala

```diff
@@ -0,0 +1,45 @@
+package io.treeverse.jpebble
+
+import java.nio.ByteBuffer
+
+import org.scalatest._
+import matchers.should._
+import funspec._
+
+class ByteBufferIndexedBytesSpec extends AnyFunSpec with Matchers {
+  val bytesSeq = Array[Byte](2, 3, 5, 7, 11, 13)
+
+  describe("ByteBufferIndexedBytes") {
+    it("returns bytes in buffer through iterator") {
+      val bytes = IndexedBytes.create(ByteBuffer.wrap(bytesSeq)).iterator.toSeq
+
+      bytes should contain theSameElementsInOrderAs bytesSeq
```

Comment:
What sorcery is this?

Reply:
It's Scala.  These tests are written in the BDD style, which traditionally use this beautiful style of [matchers](https://www.scalatest.org/user_guide/using_matchers#workingWithContainers) instead of assertions.  Ruby-like in its simplicity, unfortunately.

So... why BDD-ish specs?  Well, I wanted to be able to nest tests (`describe(...) { describe(...) { it(...) {} it(...) {..}}}`), and this was the first standard style I saw in ScalaTest that supports nesting.  Also [Google says BDD is Better](https://google.com/search?q=why+bdd+is+better).